modules/openapi-generator/src/main/resources/typescript-angular/api.service.mustache

```diff
@@ -130,6 +130,47 @@ export class {{classname}} {
 {{/operation}}
 {{/useHttpClient}}
 
+    private addToHttpParams(httpParams: HttpParams, value: any, key?: string): HttpParams {
+        if (typeof value === "object") {
+            {{#useHttpClient}}httpParams = {{/useHttpClient}}this.addToHttpParamsRecursive(httpParams, value);
+        } else {
+            {{#useHttpClient}}httpParams = {{/useHttpClient}}this.addToHttpParamsRecursive(httpParams, value, key);
+        }
+        return httpParams;
+    }
+
+    private addToHttpParamsRecursive(httpParams: HttpParams, value: any, key?: string): HttpParams {
+        if (typeof value === "object") {
+            if (this.isArrayLike(value)) {
+                (value as []).forEach( elem => httpParams = this.addToHttpParamsRecursive(httpParams, elem, key));
+            } else if (value instanceof Date) {
+                if (key != null) {
+                    {{#useHttpClient}}httpParams = {{/useHttpClient}}httpParams.append(key,
+                        (value as Date).toISOString(){{^isDateTime}}.substr(0, 10)){{/isDateTime}};
+                } else {
+                   throw Error("key may not be null if value is Date");
+                }
+            } else {
+                Object.keys(value).forEach( k => {{#useHttpClient}}httpParams = {{/useHttpClient}}this.addToHttpParamsRecursive(
+                    httpParams, value[k], key != null ? `${key}.${k}` : k));
+            }
+        } else if (key != null) {
+            {{#useHttpClient}}httpParams = {{/useHttpClient}}httpParams.append(key, value);
+        } else {
+            throw Error("key may not be null if value is not object or array");
+        }
+        return httpParams;
+    }
+
+    private isArrayLike(obj: any): boolean {
+        return (Array.isArray(obj) ||
+            typeof obj === "object" &&
+            typeof (obj.length) === "number" &&
```

Comment:
I think an actual array should be required, otherwise, the following object is also recognized as array-like, but throws an error due to the `.forEach` missing:
```typescript
{
  length: 5
  4: 'something'
}
```

Reply:
My feeling is that most JS is written with 'duck typing' in mind. Your example object _is looking for trouble_ because (duck typing aware) code could think that the object is an array. But my `isArrayLike` method will _not_ be fooled because it also checks `(obj.length - 1) in obj`. It will be fooled only with `{ length: 1, 0: 'something'}`.

I'm using here code from https://stackoverflow.com/questions/24048547/checking-if-an-object-is-array-like . In principle there are 2 option: only accepting real arrays or accept array-like objects.

If you search for 'array-like', you will find all kind of opinions to the subject. AFAIK, there are some standard objects that should be treated as arrays but aren't 'real' arrays.

Reply:
i think it should only check for `Array.isArray()`, any other array would need to be converted first
I verified my example above in the browser:
```javascript
isArrayLike({length: 5, 4: 'asdf'})
> true
```
the code should be robust for any kind of input object

Reply:
* Nope. I tried it in Firefox and it works like I said.
* There are even modules (like https://www.npmjs.com/package/@gyumeijie/is-array-like) that works like I described.
* **However, I'm easy with the Array check and will change the code**